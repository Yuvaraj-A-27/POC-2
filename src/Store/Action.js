import { ACTIVE_USER, ADD_TO_CART, CART_ACTIVE, CREATE_USER, CURRENT_PRODUCT, INITIAL_CATEGORY, INITIAL_PRODUCT, INITIAL_USER_DETAIL, LOGIN_ACTIVE, PRODUCT_POP_ACTIVE, PROFILE_ACTIVE, REGISTER_ACTIVE } from "./ActionType"

export const initialCategory = (value) =>{
    return{
        type: INITIAL_CATEGORY,
        payload : value
    }
}

export const initialProduct = (value) =>{
    return{
        type: INITIAL_PRODUCT,
        payload : value
    }
}

export const initialUserDetail = (value) =>{
    return{
        type:INITIAL_USER_DETAIL,
        payload : value
    }
}


export const loginActive = ()=>{
    return{
        type: LOGIN_ACTIVE,
    }
}

export const activeUser = (value)=>{
    return{
        type: ACTIVE_USER,
        payload: value
    }
}

export const productPopUpActive = ()=>{
    return{
        type: PRODUCT_POP_ACTIVE,
    }
}

export const currentProduct = (id) =>{
    return{
        type: CURRENT_PRODUCT,
        payload : id
    }
}

export const addToCart = (productId, userId) =>{
    return{
        type:ADD_TO_CART,
        payload: {
            productId,
            userId
        }
    }
}

export const profileActive = ()=>{
    return{
        type:PROFILE_ACTIVE,
    }
}

export const cartActive = () =>{
    return{
        type: CART_ACTIVE,
    }
}

export const registerActive = () =>{
    return{
        type : REGISTER_ACTIVE,
    }
}

export const createUser = (value) =>{
    return{
        type : CREATE_USER,
        payload : value
    }
}