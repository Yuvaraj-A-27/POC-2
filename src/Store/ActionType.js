export const INITIAL_CATEGORY = 'INITIAL_CATEGORY'
export const INITIAL_PRODUCT = 'INITIAL_PRODUCT'
export const INITIAL_USER_DETAIL = 'INITIAL_USER_DETAIL'
export const LOGIN_ACTIVE = 'LOGIN_ACTIVE'
export const ACTIVE_USER = 'ACTIVE_USER'
export const PRODUCT_POP_ACTIVE ="PRODUCT_POP_ACTIVE"
export const CURRENT_PRODUCT = 'CURRENT_PRODUCT'
export const ADD_TO_CART = 'ADD_TO_CART'
export const PROFILE_ACTIVE ='PROFILE_ACTIVE'
export const CART_ACTIVE = 'CART_ACTIVE'
export const REGISTER_ACTIVE = 'REGISTER_ACTIVE'
export const CREATE_USER = 'CREATE_USER' 
export const LOGIN_ERROR = 'LOGIN_ERROR'
export const LOGIN_ERROR_TYPING = 'LOGIN_ERROR_TYPING'