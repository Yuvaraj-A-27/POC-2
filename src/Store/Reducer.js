import { ACTIVE_USER, ADD_TO_CART, CART_ACTIVE, CREATE_USER, CURRENT_PRODUCT, INITIAL_CATEGORY, INITIAL_PRODUCT, INITIAL_USER_DETAIL, LOGIN_ACTIVE, LOGIN_ERROR, LOGIN_ERROR_TYPING, PRODUCT_POP_ACTIVE, PROFILE_ACTIVE, REGISTER_ACTIVE } from "./ActionType";

const initialState ={
    category : [],
    product : [],
    userDetail : [],
    loginActive : false,
    productPopUpActive : false,
    profileActive : false,
    cartActive : false,
    registerActive : false,
    // loginError : false,
    activeUserDetail : '',
    currentProduct : 1,
    cart:[],
}

const Reducer = (state = initialState, action)=>{
    switch(action.type){
        case INITIAL_CATEGORY: return{
            ...state,
            category : [...action.payload]
        }

        case INITIAL_PRODUCT:return{
            ...state,
            product : [...action.payload]
        }

        case INITIAL_USER_DETAIL:return{
            ...state,
            userDetail : [...action.payload]
        }

        case LOGIN_ACTIVE: return{
            ...state,
            loginActive : !state.loginActive,
            productPopUpActive : false,
            profileActive:false,
            cartActive : false,
            registerActive : false,
        }

        case PRODUCT_POP_ACTIVE: return{
            ...state,
            productPopUpActive : !state.productPopUpActive,
            loginActive : false,
            profileActive:false,
            cartActive : false,
            registerActive : false,
        }

        case ACTIVE_USER: return{
            ...state,
            activeUserDetail : action.payload
        }

        case CURRENT_PRODUCT: return{
            ...state,
            currentProduct : action.payload
        }

        case ADD_TO_CART : return{
            ...state,
            cart : [...state.cart, action.payload]
        }

        case PROFILE_ACTIVE: return{
            ...state,
            profileActive : !state.profileActive,
            loginActive : false,
            productPopUpActive:false,
            cartActive : false,
            registerActive : false,
        }

        case CART_ACTIVE: return{
            ...state,
            cartActive : !state.cartActive,
            loginActive : false,
            productPopUpActive:false,
            profileActive : false,
            registerActive : false,
        }

        case REGISTER_ACTIVE : return{
            ...state,
            registerActive : !state.registerActive,
            cartActive : false,
            loginActive : false,
            productPopUpActive:false,
            profileActive : false
        }

        case CREATE_USER : return{
            ...state,
            userDetail : [...state.userDetail, action.payload]
        }

        // case LOGIN_ERROR : return{
        //     ...state,
        //     loginError : true
        // }

        // case LOGIN_ERROR_TYPING : return{
        //     ...state,
        //     loginError : false
        // }

        default : return state
    }
}

export default Reducer;