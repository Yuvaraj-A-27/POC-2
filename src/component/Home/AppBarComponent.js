import { AppBar, Button, IconButton, makeStyles, Toolbar, Typography } from '@material-ui/core';
import React from 'react';
import  MenuIcon from '@material-ui/icons/Menu'

const useStyles = makeStyles((theme) => ({
    root: {
      flexGrow: 1,
      marginTop : '-10px',
      top: 0,
    },
    menuButton: {
      marginRight: theme.spacing(16),
    },
    title: {
      flexGrow: 1,
    },
    btn:{
        padding:'0px 10px',
        '&:hover' :{
            backgroundColor:'red',
        }
    }
  }));

function AppBarComponent({id}){
    const classes = useStyles()


    return(
        <div>
            <AppBar id={id} position="sticky" className={classes.root}>
                <Toolbar>
                <IconButton edge="start" className={classes.menuButton} color="inherit" aria-label="menu">
                    <MenuIcon />
                </IconButton>
                <Typography variant="h5" className={classes.title}>
                    MyStore
                </Typography>
                <Button className = {classes.btn} color="inherit">Register</Button>
                <Button className = {classes.btn} color="inherit">Login</Button>
                </Toolbar>
            </AppBar>
        </div>
    )
}

export default AppBarComponent;