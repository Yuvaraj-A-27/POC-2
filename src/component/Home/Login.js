import { Dialog, DialogActions, DialogContent, DialogTitle, makeStyles, TextField } from '@material-ui/core'
import React, { useState } from 'react'
import Button from '@material-ui/core/Button';
import { connect } from 'react-redux';
import { activeUser, loginActive, loginError, loginErrorTyping } from '../../Store/Action';
import axios from 'axios';
import { useHistory } from 'react-router';

const useStyles = makeStyles((theme)=>({
    margin: {
        margin: theme.spacing(1),
    },

    title:{
        margin: theme.spacing(2)
    },
    content:{
        margin: theme.spacing(3),
        marginTop:'-30px'
    },
    dialogPaper:{
        maxWidth: '400px',
        maxHeight: '1800px',
        minHeight: '300px',
    },
}))

function Login(props){
    let history = useHistory()

    //Style
    const classes = useStyles()

    //State and eventHandler
    const[userName, setUserName] = useState('')
    const[password, setPassword] = useState('')


    const loginHandler = (e)=>{
        e.preventDefault()
        if(userName && password){
            axios.post("https://fakestoreapi.com/auth/login", {
                username : userName,
                password : password
            })
            .then((res)=>{
                console.log(res.data.token);
                if(res.data.token){
                    localStorage.setItem('token',res.data.token)
                    let activeUser = props.userDetail.filter(e => e.username===userName)
                    props.activeUser(activeUser)
                    props.loginActiveAction()
                    history.push('/dashboard')
                    props.loginErrorTypingHandler()
                }
                else{
                    props.loginErrorHandler()
                }
            })
            .catch((err)=>console.log(err))
        }
        else{
            props.loginErrorHandler()
        }
        
    }

    const err = props.loginError? 'error' : null

    const userNameHandler =(e)=>{
        setUserName(e.target.value)
        props.loginErrorTypingHandler()
    }
    
    const passwordHandler = (e)=>{
        setPassword(e.target.value)
        props.loginErrorTypingHandler()
    }

    return(
    <div data-testid = {props.dataTestid}>
        <Dialog fullWidth='true' maxWidth ='lg' open={props.loginActive} onClose={props.loginActiveAction} classes={{ paperWidthLg: classes.dialogPaper }}>
            <DialogTitle className={classes.title}>Login</DialogTitle>
                <form>
                    <DialogContent className={classes.content}>
                        <TextField
                            error={err}
                            label="User Name"
                            style={{ marginBottom: '20px' }}
                            fullWidth
                            size = 'medium'
                            margin="dense"
                            variant="outlined"
                            value = {userName}
                            onChange = {userNameHandler}
                            data-testid = 'login-username'
                        />
                        <TextField
                            error={err}
                            label="Password"
                            style={{ marginTop: 5 }}
                            fullWidth
                            type = 'password'
                            size = 'medium'
                            margin="dense"
                            variant="outlined"
                            value = {password}
                            onChange = {passwordHandler}
                        />
                        
                    </DialogContent>
                    <DialogActions>
                        <Button 
                            variant="contained" 
                            size="medium" 
                            color="primary" 
                            className={classes.margin}
                            onClick = {loginHandler}
                        >Login
                        </Button>
                    </DialogActions>
                </form>
        </Dialog>
    </div>
    )
}

const mapStateToProps = state =>{
    return{
        loginActive : state.loginActive,
        userDetail : state.userDetail,
        loginError : state.loginError
    }
}

const mapDispatchToProps = dispatch =>{
    return{
        loginActiveAction : ()=>dispatch(loginActive()),
        activeUser : (value)=> dispatch(activeUser(value)),
        loginErrorHandler : ()=>dispatch(loginError()),
        loginErrorTypingHandler : () =>dispatch(loginErrorTyping())
    }
}

export default connect(mapStateToProps,mapDispatchToProps)(Login)