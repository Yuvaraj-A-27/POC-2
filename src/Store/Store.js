import { createStore } from "redux"
import persistReducer from "redux-persist/es/persistReducer"
import persistStore from "redux-persist/es/persistStore"
import Reducer from "./Reducer"
import storage from 'redux-persist/lib/storage'


const persistConfig = {
    key:'root',
    storage,
}

const persistorReducer = persistReducer(persistConfig, Reducer)

export const Store = createStore(persistorReducer)
export const Persistor = persistStore(Store)