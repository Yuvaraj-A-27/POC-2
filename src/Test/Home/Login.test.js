import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { fireEvent, render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Login from '../../component/Home/Login'
import AppBarComponent from '../../component/Home/AppBarComponent'
import { LOGIN_ACTIVE } from '../../Store/ActionType'


const middleware  = {}
const mockStore = configureStore(middleware)

const initialState = {}
const Store = mockStore(initialState)


describe('Login Component Test', ()=>{

    test('Login Functionality Test',()=>{

        const {getByTestId} = render(
            <Provider store={Store}>
                <AppBarComponent />
                <Login />
            </Provider>
        )

        const loginActive = () => ({type: LOGIN_ACTIVE})
        Store.dispatch(loginActive())
        expect(Store.getActions()).toEqual([{type:LOGIN_ACTIVE}])

        const login = renderer.create(
            <Provider store={Store}>
                {/* <AppBarComponent/> */}
                <Login />
            </Provider>
        ).toJSON()
        expect(login).toMatchSnapshot()

        
        // const username = getByTestId('login-username')
        // fireEvent.change(username,{target:{value:'Yuvaraj'}})
        // expect(username.value).toBe('Yuvaraj')
    })

    test('Login Component snapShot test', ()=>{
        const login = renderer.create(
            <Provider store={Store}>
                {/* <AppBarComponent/> */}
                <Login />
            </Provider>
        ).toJSON()
        expect(login).toMatchSnapshot()
    })

})