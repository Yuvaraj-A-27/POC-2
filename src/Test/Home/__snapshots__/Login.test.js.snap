// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component Test Login Component snapShot test 1`] = `<div />`;
